tasks/change-granule-collection-s3/tests/test-index.js

```diff
@@ -0,0 +1,1000 @@
+  t.context.systemBucket = randomId('system');
+  t.context.stackName = 'changeGranuleCollectionS3TestStack';
+  const bucketMapping = {
+    public: t.context.publicBucket,
+    protected: t.context.protectedBucket,
+    private: t.context.privateBucket,
+    system: t.context.systemBucket,
+  };
+  t.context.bucketMapping = bucketMapping;
+  await Promise.all([
+    s3().createBucket({ Bucket: t.context.publicBucket }),
+    s3().createBucket({ Bucket: t.context.protectedBucket }),
+    s3().createBucket({ Bucket: t.context.privateBucket }),
+    s3().createBucket({ Bucket: t.context.systemBucket }),
+  ]);
+  process.env = {
+    ...process.env,
+    PG_DATABASE: testDbName,
+    DISTRIBUTION_ENDPOINT: 'https://something.api.us-east-1.amazonaws.com/',
+    system_bucket: t.context.systemBucket,
+    stackName: t.context.stackName,
+  };
+  await putJsonS3Object(
+    t.context.systemBucket,
+    getDistributionBucketMapKey(t.context.stackName),
+    {
+      [t.context.publicBucket]: t.context.publicBucket,
+      [t.context.privateBucket]: t.context.privateBucket,
+      [t.context.protectedBucket]: t.context.protectedBucket,
+      [t.context.systemBucket]: t.context.systemBucket,
+    }
+  );
+});
+
+test.afterEach.always(async (t) => {
+  await recursivelyDeleteS3Bucket(t.context.privateBucket);
+  await recursivelyDeleteS3Bucket(t.context.publicBucket);
+  await recursivelyDeleteS3Bucket(t.context.protectedBucket);
+  await recursivelyDeleteS3Bucket(t.context.systemBucket);
+});
+
+test.serial('Should move files to final location with cmr xml file', async (t) => {
+  const payloadPath = path.join(__dirname, 'data', 'payload_cmr_xml.json');
+  t.context.payload = JSON.parse(fs.readFileSync(payloadPath, 'utf8'));
+  const filesToUpload = granulesToFileURIs(
+    t.context.payload.input.granuleIds, t
+  );
+  const collection = { name: 'MOD11A1', version: '002' };
+  const newPayload = buildPayload(t, collection);
+  await uploadFiles(filesToUpload, t.context.bucketMapping);
+  const output = await changeGranuleCollectionS3(newPayload);
+  await validateOutput(t, output);
+  t.true(await s3ObjectExists({
+    Bucket: t.context.protectedBucket,
+    Key: 'example2/2003/MOD11A1.A2017200.h19v04.006.2017201090724.hdf',
+  }));
+  t.true(await s3ObjectExists({
+    Bucket: t.context.publicBucket,
+    Key: 'jpg/example2/MOD11A1.A2017200.h19v04.006.2017201090724_1.jpg',
+  }));
+  t.true(await s3ObjectExists({
+    Bucket: t.context.publicBucket,
+    Key: 'example2/2003/MOD11A1.A2017200.h19v04.006.2017201090724_2.jpg',
+  }));
+  t.true(await s3ObjectExists({
+    Bucket: t.context.publicBucket,
+    Key: 'example2/2003/MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml',
+  }));
+  const UMM = await metadataObjectFromCMRFile(
+    `s3://${t.context.publicBucket}/example2/2003/` +
+    'MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml'
+  );
+
+  const CollectionInformation = UMM.Granule.Collection;
```

Comment:
It's probably a minor quibble, but we're not directly testing that it *only* changed this value.     Thoughts?